src/download_fossa_cli.rs

```diff
@@ -0,0 +1,191 @@
+    let path = latest_release_response.url().path();
+
+    let tag = path
+        .rsplit('/')
+        .next()
+        .ok_or(Error::FindingVersion)
+        .into_report()
+        .describe_lazy(|| format!("Parsing fossa-cli version from path {}", path))?;
+
+    if !tag.starts_with('v') {
+        return Err(Error::FindingVersion)
+            .into_report()
+            .describe_lazy(|| format!(r#"Expected tag to start with v, but found "{tag}""#))?;
+    }
+    let version = tag.trim_start_matches('v');
+
+    // currently supported os/arch combos:
+    // darwin/amd64
+    // linux/amd64
+    // windows/amd64
+    //
+    // We only support "amd64" right now, so no need to look at std::env::consts::ARCH
+    let arch = "amd64";
+    let extension = ".zip";
+    let os = match std::env::consts::OS {
+        "macos" => "darwin",
+        "windows" => "windows",
+        _ => "linux",
+    };
+
+    // Example URLs:
+    // https://github.com/fossas/fossa-cli/releases/download/v3.7.2/fossa_3.7.2_darwin_amd64.zip
+    // https://github.com/fossas/fossa-cli/releases/download/v3.7.2/fossa_3.7.2_linux_amd64.zip
+    let download_url = format!("https://github.com/fossas/fossa-cli/releases/download/v{version}/fossa_{version}_{os}_{arch}{extension}");
+    let content = download_from_github(download_url).await?;
+
+    let final_path = config_dir.join(command_name());
+    unzip_zip(content, &final_path).await?;
+    Ok(final_path)
+}
+
+#[tracing::instrument]
+async fn download_from_github(download_url: String) -> Result<Cursor<Bytes>, Error> {
+    let client = reqwest::Client::new();
+    let response = client
+        .get(&download_url)
+        .send()
+        .await
+        .into_report()
+        .change_context(Error::Downloading)
+        .describe_lazy(|| {
+            format!("Error while downloading latest fossa release from {download_url}")
+        })?;
+
+    let content = response
+        .bytes()
+        .await
+        .into_report()
+        .change_context(Error::Downloading)
+        .describe_lazy(|| {
+            format!(
+                "Error while converting download of fossa release from {download_url} into bytes"
+            )
+        })?;
+    let content = Cursor::new(content);
+    Ok(content)
+}
+
+#[tracing::instrument(skip(content))]
+async fn unzip_zip(content: Cursor<Bytes>, final_path: &Path) -> Result<(), Error> {
+    let mut archive = zip::ZipArchive::new(content)
+        .context(Error::Extracting)
+        .describe("extracting zip file from downloaded fossa release")?;
+    let mut file = match archive.by_name("fossa") {
+        Ok(file) => file,
+        Err(..) => {
+            return Err(Error::Extracting)
+                .into_report()
+                .describe("finding fossa executable in downloaded fossa release");
+        }
+    };
+
+    let mut final_file = fs::OpenOptions::new()
+        .create(true)
+        .write(true)
+        .mode(0o770)
```

Comment:
`.mode` [is implemented on](https://doc.rust-lang.org/std/os/unix/fs/trait.OpenOptionsExt.html#tymethod.mode) `std::os::unix::fs::OpenOptionsExt` and won't work on Windows (which has no concept of unix permission modes).
Do you really need to set this to be non-default permissions?

Reply:
In Windows, whether it is executable is just determined by file extension. You probably just need to do this in three steps:

1. Download the zip
2. Extract the zip to a temp location
3. Use a conditionally compiled function that moves the binary from the temp location to its destination, and on unix sets it executable (make sure to `use std::os::unix::fs::OpenOptionsExt;` inside the function, not at the top of the file).